src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -1471,14 +1487,11 @@ abstract class ManagedStrategy(
             loopDetector.passParameters(params)
         }
         // notify loop detector about the method call
-        if (guarantee == null) {
+        if (section == AnalysisSectionType.NORMAL) {
             loopDetector.beforeMethodCall(codeLocation, params)
         }
-        // if the method is atomic or should be ignored, then we enter an ignored section
-        if (guarantee == ManagedGuaranteeType.IGNORE ||
-            guarantee == ManagedGuaranteeType.ATOMIC) {
-            enterIgnoredSection()
-        }
+        // if the method has certain guarantees, enter the corresponding section
+        enterAnalysisSection(threadId, section)
```

Comment:
What is `section` here? Please explain in the comments.

Reply:
There is already a comment:

https://github.com/JetBrains/lincheck/blob/dba37f265dc8d18c03d44221ae040ea4a4354f19/src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt#L1455

And I renamed `section` to `methodSection` for clarity